tests/test_voting_classifier.py

```diff
@@ -261,3 +300,108 @@ def test_binary_hard_num_agree():
     X = ["mock", "data", "not used"]
     Y = voting_classifier.predict(X)
     assert np.array_equal(Y, Y_expected)
+
+def test_multiclass_hard():
+
+    Y1_prob = np.array([
+        [0.6, 0.1, 0.1, 0.2],
+        [0.2, 0.3, 0.1, 0.4],
+        [0.3, 0.1, 0.4, 0.2]
+    ])
+    Y2_prob = np.array([
+        [0.5, 0.3, 0.2, 0.0],
+        [0.1, 0.6, 0.1, 0.2],
+        [0.1, 0.3, 0.1, 0.5]
+    ])
+    Y3_prob = np.array([
+        [0.5, 0.3, 0.2, 0.0],
+        [0.2, 0.3, 0.1, 0.4],
+        [0.1, 0.5, 0.1, 0.3]
+    ])
+    # Using highest values:
+    # 1st data point: 0: 3 votes -> clear winner is 0
+    # 2nd data point: 1: 1 vote, 3: 2 votes -> clear winner (in majority voting) is 3
+    # 3rd data point: 2: 1 vote, 3: 1 vote, 1: 1 vote -> no winner, pick first in numerical order, so 1
+    Y_expected = np.array([
+        0,
+        3,
+        1
+    ])
+    est1 = MockEstimator(Y1_prob)
+    est2 = MockEstimator(Y2_prob)
+    est3 = MockEstimator(Y3_prob)
+
+    voting_classifier = WellcomeVotingClassifier(
+        estimators=[est1, est2, est3], voting="hard"
+    )
+    X = ["mock", "data", "not used"]
+    Y = voting_classifier.predict(X)
+
+    assert np.array_equal(Y, Y_expected)
+
+def test_multiclass_hard_num_agree():
+
+    Y1_prob = np.array([
+        [0.6, 0.1, 0.1, 0.2],
+        [0.2, 0.3, 0.1, 0.4],
+        [0.3, 0.1, 0.4, 0.2]
+    ])
+    Y2_prob = np.array([
+        [0.5, 0.3, 0.2, 0.0],
+        [0.1, 0.6, 0.1, 0.2],
+        [0.1, 0.3, 0.1, 0.5]
+    ])
+    Y3_prob = np.array([
+        [0.5, 0.3, 0.2, 0.0],
+        [0.2, 0.3, 0.1, 0.4],
+        [0.1, 0.5, 0.1, 0.3]
+    ])
+    # Using highest values:
+    # 1st data point: 0: 3 votes -> clear winner is 0
+    # 2nd data point: 1: 1 vote, 3: 2 votes -> no winner, pick first in numerical order, so 1
+    # 3rd data point: 2: 1 vote, 3: 1 vote, 1: 1 vote -> no winner, pick first in numerical order, so 1
+    Y_expected = np.array([
+        0,
+        1,
+        1
+    ])
+    est1 = MockEstimator(Y1_prob)
+    est2 = MockEstimator(Y2_prob)
+    est3 = MockEstimator(Y3_prob)
+
+    voting_classifier = WellcomeVotingClassifier(
+        estimators=[est1, est2, est3],
+        voting="hard",
+        num_agree=3
+    )
+    X = ["mock", "data", "not used"]
+    Y = voting_classifier.predict(X)
+    assert np.array_equal(Y, Y_expected)
+
+def test_inc_vectorizer():
```

Comment:
took me some seconds to realise what was the test here. maybe rename to `test_pipeline_with_vectorizer`